server/src/main/java/org/elasticsearch/index/mapper/IndexFieldMapper.java

```diff
@@ -189,6 +191,12 @@ public Query wildcardQuery(String value,
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
             return new ConstantIndexFieldData.Builder(mapperService -> fullyQualifiedIndexName);
         }
+
+        @Override
+        public ValuesSourceType getValuesSourceType() {
+            // TODO: Should Index fields be aggregatable?  What even is an IndexField?
```

Comment:
It's an `_index` sorta-metadata field that we add to the doc, mainly for use with querying to differentiate between different indices.  But I've seen it used in aggs before, like Terms, to partition based on index

https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index-field.html